app/src/main/java/com/thanksmister/iot/mqtt/alarmpanel/ui/views/BaseAlarmView.kt

```diff
@@ -101,9 +106,53 @@ abstract class BaseAlarmView : LinearLayout {
         }
     }
 
+    override fun onVisibilityChanged(changedView: View?, visibility: Int) {
+        super.onVisibilityChanged(changedView, visibility)
+
+        if (!this.isShown){
+            stopFingerprintIdentity()
+            return
+        }
+
+        startFingerprintIdentity()
```

Comment:
I have added a check for Android M here so we do not bother using the FingerprintManager when the hardware or fingerprints are not available or the device doesn't support this feature.   In the future i would like to add the option to enable/disable fingerprint unlock and update the UI so that it shows a fingerprint scanner view.  If user fails fingerprint scan, then the view reverts to code entry to enter the backup code.